ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,376 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc(MSFBottomSheetControllerDelegate)
+public protocol BottomSheetControllerDelegate: AnyObject {
+    /// Called after the sheet fully expanded.
+    @objc optional func bottomSheetControllerDidExpand(_ controller: BottomSheetController)
+
+    /// Called after the sheet fully collapsed.
+    @objc optional func bottomSheetControllerDidCollapse(_ controller: BottomSheetController)
+}
+
+@objc(MSFBottomSheetController)
+public class BottomSheetController: UIViewController {
+
+    /// Initializes the bottom sheet controller.
+    /// - Parameter contentViewController: The view controller that manages the bottom sheet content.
+    ///
+    /// By default the root view of `contentViewController` will be sized automatically to fill the available area,
+    /// respecting the provided `preferredExpandedHeightFraction` multiplier.
+    /// Alternatively, the content can size itself by setting `respectsPreferredContentSize` to true and providing a `preferredContentSize`.
+    @objc public init(with contentViewController: UIViewController) {
+        self.contentViewController = contentViewController
+        super.init(nibName: nil, bundle: nil)
+
+        addChild(contentViewController)
+        contentViewController.didMove(toParent: self)
```

Comment:
I may be off base with this, but having the view controller hierarchy management live separate of the view hierarchy management (done in a lazy getter indirectly in loadView) seems scary to me. Can we avoid doing any view controller hierarchy manipulation until someone actually requests our view? And then the bonus is viewController and viewHierarchy manipulation are in the same place and less likely to mysteriously go out of sync?

Reply:
Can you elaborate on what's scary? My thinking is that once it's our child VC, we can do as we please with the root view, which includes delaying when that view is added to any hierarchy. Is there an expectation that the view will be moved immediately after addChild is called?

Reply:
We've run into many issues over the years with the view controller hierarchy not correctly reflecting the view hierarchy which leads to subtle bugs with view controller lifecycle events not propagating and stuff like that. Having it obvious they are paired in code has helped prevent causing more of those bugs with refactoring